Added changes to git ignore

kingce opened the pull request:
Added files to the .gitignore so it doesn't track changes to the files generated by the install/build process. Lmk if this is a bad idea, I was mostly just tired of 'git status' telling me to track files that we're not really planning on tracking.

kingce committed:
added changes to git ignore

kingce commented:
** Sorry for spam, not sure if i'm supposed to invite the entire team or just 1-2 devs for PR's. Figured since the dev team is pretty small might as well just invite everyone so no one's left out of the loop.

clemwj commented on .gitignore:
```diff
@@ -1,2 +1,32 @@
 *.exe
 .vs/
+
```
While a good idea, this addresses a symptom and not the cause. Instead you should add a clean target to the Makefile and get autotools to clean up after itself. Doing those two things will make it so the amount of files is minimal/easier to clean.

kingce committed:
added remove targets for all files generated by autotools/not tracked by git

clemwj commented on program/Makefile:
```diff
@@ -1,2 +1,21 @@
 	valac -o PDFGrader.exe --pkg=gtk+-3.0 --pkg=poppler-glib --pkg cairo --pkg gee-0.8 --pkg glib-2.0 --pkg libxml-2.0 --pkg posix src/main/System.vala src/main/QuestionSet.vala src/main/Question.vala src/main/Mark.vala src/main/Save.vala src/main/Lock.vala src/main/Export.vala
+
+clean: 
+	rm -f Makefile.in
```
Looks like you missed the *.c and *.o files. Also I would see what you can do to get autotools to clean up after itself. if you aren't able to get it working, I would suggest adding another target like 'clean-autotools' and remove all the generated autotools files minus the .tar and .exe

clemwj replied:
And use it at the end of maintain.sh

kingce committed:
tentatively made clean-autotools target

kingce committed:
added autotools generated Makefile to the clean-autotools target

kingce committed:
made changes to makefile process

kingce commented:
Okay so turns out it wasn't as simple as changing the name to lowercase m, since this fucks with autotool's vibes or whatever (our makefile takes precedence over the generated Makefile, meaning it couldn't run a few of the things necessary to build the project). Also we can't add clean-autotools to the end of ./maintain.sh since that fucks with the ability to call 'make' while developing. 

Since our own custom Makefile only consists of the PDFGrader.exe target, I just added that and the custom clean targets (clean-autotools, clean-custom, etc) to the Makefile.am, which is what autotools uses to generate the Makefile that was overwriting ours. This way our custom targets get added and we can use them no prob. 

Since the whole purpose of this branch is to simplify workflow for development, I added a 'clean-git' target that calls 'clean-autools' and 'clean-custom'(removes all the other non autotools generated files that we don't want tracked by git). So the dev workflow from start to finish would basically be 

./maintain.sh
do some work, call make whenever
work is done -> make clean-git 
git commit/push

I propose we stop tracking the Makefile in git then, since it will be generated by autotools when maintain.sh is run. 

Lmk if there are any potential issues with this
